postcommit.py

```diff
@@ -0,0 +1,642 @@
+        for diff_content in diff:
+            file_name_pattern = re.compile('^\\+\\+\\+ b/([^\\s]+)', re.M)
+            modified_lines_pattern = re.compile('^@@[-0-9,\\s]+\\+([0-9+,]+)\\s*@@$', re.M)
+
+            if not file_name:
+                file_name = file_name_pattern.findall(diff_content)
+            if not match_values:
+                match_values = modified_lines_pattern.findall(diff_content)
+
+            if file_name and match_values:
+                file_name = file_name[0].replace('/', os.sep).strip()
+                diff_changes.append({'file': file_name, 'changed_lines': self.getDiffLines(match_values)})
+                file_name = match_values = []
+
+        return diff_changes
+
+# 	***************************************************************************************************************
+#	difflist :
+#
+#			Try to get diff files list between two changesets
+#
+#			@param : rev1
+#			@param : rev2
+#			@param : files
+#			
+#			@return : [('README', 1, 0, False), ('testfile', 1, 0, False)]
+#       						
+#	****************************************************************************************************************
+
+    def difflist(self, rev1, rev2, files=[], ignore_space=False, *pats, **opts):
+        opts['stat'] = True
+ 
+        if ignore_space:
+            opts['ignore_space_change'] = True
+            opts['ignore_all_space'] = True
+            opts['ignore_blank_lines'] = True
+ 
+        opts['rev'] = [str(rev1)+":"+str(rev2), ]
+        diffopts = patch.diffopts(self.ui, opts)
+        if files:
+            file_list = (files,[],[])
+            diff_content = patch.diff(self.repo, rev1, rev2, changes=file_list, opts=diffopts)
+        else:
+            diff_content = patch.diff(self.repo, rev1, rev2, opts=diffopts)
+        diff_fileslist = patch.diffstatdata(util.iterlines(diff_content))
+        return diff_fileslist
+
+ 
+# 	***************************************************************************************************************
+#	diffstat :
+#
+#			Return a files added,removed lines
+#
+#			@param : rev1
+#			@param : rev2
+#			@param : files
+#			
+#			@return :{'AddedLines': 1, 'File': 'testfile', 'RemovedLines': 0}
+#       						
+#	****************************************************************************************************************
+    def diffstat(self, rev1, rev2, files, *pats, **opts):
+	if files:
+        	diff_fileslist = self.difflist(rev1, rev2, files, *pats, **opts)
+		file_stat=[]
+		for x in diff_fileslist:
+			file_numstats = { 'File':[],'AddedLines':[],'RemovedLines':[] }
+			file_numstats['File']=x[0]
+			file_numstats['AddedLines']=x[1]
+			file_numstats['RemovedLines']=x[2]
+			file_stat.append(file_numstats)
+		return file_stat
+	return []
+        
+
+
+class GitRepository():
+
+	def __init__(self, local_repository):
+		self.logs = []
+		self.module_dir = local_repository
+        	self.oldId = oldId
+        	self.newId = newId
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Rename this field "newId" to match the regular expression ^[_a-z][_a-z0-9]*$. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=python%3AS116)